### DISCUSSION TITLE
Makefile error

### DISCUSSION THREAD

**Discussion by ShubhSensei** [Cyfrin/foundry-full-course-cu#291] [Q&A]
when i run command this command in terminal, it works
```command
forge script script/deployFundMe.s.sol --rpc-url $RPC_URL --private-key $PRIVATE_KEY --broadcast
```

but when I enter the same command in Makefile 

<img width="761" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91625601/7875c283-73e8-473a-b15b-d1f2ac8723db">

& run `make deploy`, it give me error
```error
Error: 
Failed to create wallet from private key. Private key is invalid hex: Invalid character 'R' at position 0

Context:
- Invalid character 'R' at position 0
make: *** [Makefile:4: deploy] Error 1
```

so I changed the Makefile command to 

<img width="748" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91625601/7e2532a9-16ae-4964-8d5f-a62e975c92ba">

but still it gives me this error
```
SIMULATION COMPLETE. To broadcast these transactions, add --broadcast and wallet configuration(s) to the previous command. See forge script --help for more.

Transactions saved to: /mnt/d/foundry/FundMe/broadcast/deployFundMe.s.sol/31337/dry-run/run-latest.json

Sensitive values saved to: /mnt/d/foundry/FundMe/cache/deployFundMe.s.sol/31337/dry-run/run-latest.json

/bin/sh: 1: --private-key: not found
/bin/sh: 1: --broadcast: not found
make: *** [Makefile:4: deploy] Error 127
```

Also, I crossed-checked the keys in .env file, is correct.

**Comment by shikhar229169** [comment]
So the below line is displayed when you don't add --broadcast at the end
```
To broadcast these transactions, add --broadcast and wallet configuration(s) to the previous command.
```
But as I can see you have it written in both commands.
My guess is that you have not saved your files.

**Reply by ShubhSensei** [reply]
Like don't they save automatically? What about 'source .env', I always run it in terminal.

**Reply by shikhar229169** [reply]
source .env has no effect for MakeFile. Because it directly takes value from env file.

**Reply by shikhar229169** [reply]
Can you remove that @ sign from the starting, and tell whether the output you are getting is correct.
After removing @ sign, when you run the deploy command via make, it will echo the command by putting all the actual values from env file and then run it.

**Reply by cromewar** [reply]
@SensationalShubham does @shikhar229169 solution worked for you?
